onboard/gps/src/__main__.py

```diff
@@ -0,0 +1,162 @@
+import serial
```

Comment:
Can you add a docstring to the top of the file and for each function explaining what is going on from a high level or what is the purpose of the file/class/function?